src/witan/workspace_api.clj

```diff
@@ -65,9 +59,37 @@
              (s/one s/Keyword "predicate-true")
              (s/one s/Keyword "predicate-false")]) "egress")])
 
-(def WorkflowModel
+(def CatalogEntry
+  {:witan/name    s/Keyword
+   :witan/fn      s/Keyword
+   :witan/version s/Str
+   (s/optional-key :witan/params) {s/Keyword s/Any}})
+
+(def ModelMetaData
+  "Schema for the Witan workflow model metadata"
+  {:witan/name          s/Keyword
+   :witan/version       s/Str
+   :witan/doc           s/Str})
+
+(defn model-valid?
+  [{:keys [workflow catalog]}]
+  (try
+    (let [node-names    (-> workflow flatten set)
+          catalog-names (set (map :witan/name catalog))
+          groups        (group-by :witan/name catalog)]
+      (cond
+        (not-empty (clojure.set/difference node-names catalog-names))
+        (do (println "An error occurred: There are missing :witan/name entries in the catalog.")
```

Comment:
Haha, yeah, this is this only obvious way to get information out of a schema constraint predicate. I shall remove the `do`s but...these will only occur at compile time (and will prevent compilation) so...acceptable?